parser-typechecker/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -132,10 +207,11 @@ generateConstructorMapping oldComponent newComponent = Map.fromList
 propagate
   :: forall m i v
    . (Applicative m, Var v)
-  => Patch
+  => Names0
+  -> Patch
   -> Branch0 m
   -> F m i v (Edits v)
-propagate patch b = case validatePatch patch of
+propagate rootNames patch b = case validatePatch patch of
```

Comment:
Okay, I did some rejiggering of this. 

I left `Names0` alone as a dependency but pulled all usage of it out into a separate function with a big comment that it can be deleted once patches more natively support constructor mapping. 

I could also have passed in like, a pair of functions for looking up term and type names but that didn't seem worth it to me given that the usage of `Names0` only pertains to code that will hopefully be deleted soon anyway.

Reply:
Also just added an explicit export list to `Propagate.hs` - it's literally one function, `propagateAndApply`.